tests/testthat/test-as_gt.R

```diff
@@ -0,0 +1,290 @@
+skip_on_cran()
+
+my_tbl_summary <- trial |>
+  select(trt, age, death) |>
+  tbl_summary() |>
+  modify_table_styling(columns = label, footnote = "test footnote", rows = variable == "age")
+my_tbl_cross <- tbl_cross(trial, trt, grade)
+my_tbl_regression <- lm(marker ~ age, trial) |> tbl_regression()
+my_tbl_uvregression <- trial |> tbl_uvregression(method = lm, y = age)
+my_spanning_tbl <- trial |>
+  tbl_summary(by = grade, include = age) |>
+  modify_spanning_header(c(stat_1, stat_3) ~ "**Testing**")
+
+expect_silent(gt_tbl_summary <- my_tbl_summary |> as_gt())
+expect_silent(gt_tbl_cross <- my_tbl_cross |> as_gt())
+expect_silent(gt_tbl_regression <- my_tbl_regression |> as_gt())
+expect_silent(gt_tbl_uvregression <- my_tbl_uvregression |> as_gt())
+expect_silent(gt_spanning_tbl <- my_spanning_tbl |> as_gt())
+
+test_that("as_gt works with standard use", {
+  expect_silent(my_tbl_summary |> as_gt(return_calls = TRUE))
+  expect_silent(my_tbl_summary |> as_gt(include = gt))
+  expect_equal(
+    names(gt_tbl_summary),
+    c("_data", "_boxhead", "_stub_df", "_row_groups", "_heading", "_spanners", "_stubhead", "_footnotes",
+      "_source_notes", "_formats", "_substitutions", "_styles", "_summary", "_options", "_transforms",
+      "_locale", "_has_built")
+  )
+})
+
+test_that("as_gt passes table body correctly", {
+  # tbl_summary
+  expect_equal(
+    my_tbl_summary$table_body,
+    gt_tbl_summary$`_data`
+  )
+
+  # tbl_cross
+  expect_equal(
+    my_tbl_cross$table_body,
+    gt_tbl_cross$`_data`
+  )
+
+  # tbl_regression
+  expect_equal(
+    my_tbl_regression$table_body |>
+      dplyr::mutate(conf.low = paste0(round(conf.low, digits = 2), ", ", round(conf.high, digits = 2))),
+    gt_tbl_regression$`_data`,
+    ignore_attr = "class"
+  )
+
+  # tbl_uvregression
+  expect_equal(
+    my_tbl_uvregression$table_body |>
+      dplyr::select(-conf.low),
+    gt_tbl_uvregression$`_data` |>
+      dplyr::select(-conf.low),
+    ignore_attr = "class"
+  )
```

Comment:
Since `conf.low` is unformatted in the `gtsummary` `table_body` and all formatting is applied within the `gt` `_data` object these two columns don't match.  Is there a function like `apply_fmt_fn` in `cards` that applies formatting functions and/or column merging formatting to gtsummary `table_body` objects? If so that would also be helpful for adding tests for formatting functions since I don't want to overcomplicate if possible.

Reply:
In this case, could you use `as_tibble()` or `as.data.frame()`? Those apply all the formatting functions, merging, etc. If you need more columns in `.$table_body` for testing, you can can run `modify_column_unhide()` and they'll be kept when converted to a tibble.